modules/stats.php

```diff
@@ -1754,7 +1757,7 @@ function stats_get_from_restapi( $args = array(), $resource = '' ) {
 	// Do the dirty work.
 	$response = Jetpack_Client::wpcom_json_api_request_as_blog( $endpoint, $api_version, $args );
 	if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
-		$data = array();
+		$data = $response;
```

Comment:
`$response` does not have to be a `WP_Error` for every non-200 response.

I think we should do something like:
```
$data = is_wp_error( $response ) ? $response : new WP_Error( 'stats_error' );
```

Reply:
That makes sense. Before 7.1, we just returned the response, but there could be unexpected cases so we should force the error. 👍 

https://github.com/Automattic/jetpack/pull/11582